instructor/common.py

```diff
@@ -1050,7 +1061,27 @@ def facs_histograms_edit(request, assignment, sample_prep, sp):
 
 def preview(request, assignment):
     a = models.Assignment.objects.get(id=assignment)
-    return render_to_response('instructor/preview.html',
-                              {'assignment': a,
-                               'assignment_json': compiler.preview_as_json(a.id)},
-                              context_instance=RequestContext(request))
+
+    if is_assignment_complete(a):
+        return render_to_response(
+            'instructor/preview.html',
+            {
+                'assignment': a,
+                'assignment_json': compiler.preview_as_json(a.id)
+            },
+            context_instance=RequestContext(request))
+    else:
+        error = "You must define your experimental setup and the variables " \
+                "for at least one experimental technique before you can preview " \
+                "the assignment."
+        return redirect('common_assignments', error=error)
```

Comment:
I get this error when I get to this case:

    Reverse for 'common_assignments' with arguments '()' and keyword arguments
     '{'error': 'You must define your experimental setup and the variables for at least
     one experimental technique before you can preview the assignment.'}' not found. 
    1 pattern(s) tried: ['ab/assignments/$']